contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol

```diff
@@ -0,0 +1,1456 @@
+        markets via multiple separate extension contracts, `totalCollateralValue` should be counted only once.
+        */
+        int256 totalCollateralValue = _actionInfo.accountInfo.collateralBalance
+            .add(_actionInfo.accountInfo.owedRealizedPnl)
+            .add(_actionInfo.accountInfo.pendingFundingPayments);
+
+        // Note: Both basePositionValue and quoteValue are values that belong to the single market managed by this contract.
+        int256 unrealizedPnl = _actionInfo.basePositionValue.add(_actionInfo.quoteValue);
+
+        int256 accountValue = totalCollateralValue.add(unrealizedPnl);
+
+        if (accountValue <= 0) {
+            return 0;
+        }
+
+        // `accountValue` is always positive. Do not use absolute value of basePositionValue in the below equation,
+        //  to keep the sign of CLR same as that of basePositionValue.
+        return _actionInfo.basePositionValue.preciseDiv(accountValue);
+    }
+
+    /**
+     * Calculate the new leverage ratio. The methodology reduces the size of each rebalance by weighting
+     * the current leverage ratio against the target leverage ratio by the recentering speed percentage. The lower the recentering speed, the slower
+     * the leverage token will move towards the target leverage each rebalance.
+     *
+     * return int256          New leverage ratio
+     */
+    function _calculateNewLeverageRatio(int256 _currentLeverageRatio) internal view returns(int256) {
+        // Convert int256 variables to uint256 prior to passing through methodology
+        uint256 currentLeverageRatioAbs = _currentLeverageRatio.abs();
+        uint256 targetLeverageRatioAbs = methodology.targetLeverageRatio.abs();
+        uint256 maxLeverageRatioAbs = methodology.maxLeverageRatio.abs();
+        uint256 minLeverageRatioAbs = methodology.minLeverageRatio.abs();
+
+        // CLRt+1 = max(MINLR, min(MAXLR, CLRt * (1 - RS) + TLR * RS))
+        // a: TLR * RS
+        // b: (1- RS) * CLRt
+        // c: (1- RS) * CLRt + TLR * RS
+        // d: min(MAXLR, CLRt * (1 - RS) + TLR * RS)
+        uint256 a = targetLeverageRatioAbs.preciseMul(methodology.recenteringSpeed);
+        uint256 b = PreciseUnitMath.preciseUnit().sub(methodology.recenteringSpeed).preciseMul(currentLeverageRatioAbs);
+        uint256 c = a.add(b);
+        uint256 d = Math.min(c, maxLeverageRatioAbs);
+        uint256 newLeverageRatio = Math.max(minLeverageRatioAbs, d);
+
+        return _currentLeverageRatio > 0 ? newLeverageRatio.toInt256() : newLeverageRatio.toInt256().neg();
+    }
+
+    /**
+     * Calculate total notional rebalance quantity and chunked rebalance quantity in base asset units for engaging the SetToken. Used in engage().
+     * Leverage ratio (for the base asset) is zero before engage. We open a new base asset position with size equals to ( (collateralBalance/2) * targetLeverageRatio / baseAssetPrice)
```

Comment:
Only a nit but a lot of javadoc comments in this contract are quite wide and you have scroll sideways to read them on Github on a laptop. Maybe we could pick a standard col width (120?) to keep things easily legible.

Reply:
Limited to 160. Trying not to introduce unnecessary diffs between this and PerV2LeverageStrategyExtension contract.